ecmascript/transforms/module/src/common_js.rs

```diff
@@ -17,25 +20,56 @@ use swc_ecma_utils::IsDirective;
 use swc_ecma_utils::{var::VarCollector, DestructuringFinder, ExprFactory};
 use swc_ecma_visit::{noop_fold_type, Fold, FoldWith, VisitWith};
 
-pub fn common_js(root_mark: Mark, config: Config) -> impl Fold {
+pub fn common_js(
+    root_mark: Mark,
+    config: Config,
+    scope_arg: Option<Rc<RefCell<Scope>>>,
+) -> impl Fold {
+    let scope: Rc<RefCell<Scope>> = match scope_arg {
+        Some(scope) => scope,
+        None => Rc::new(RefCell::new(Default::default())),
+    };
     CommonJs {
         root_mark,
         config,
-        scope: Default::default(),
+        scope,
         in_top_level: Default::default(),
     }
 }
 
 struct CommonJs {
     root_mark: Mark,
     config: Config,
-    scope: Scope,
+    scope: Rc<RefCell<Scope>>,
     in_top_level: bool,
 }
 
 impl Fold for CommonJs {
     noop_fold_type!();
 
+    fn fold_module(&mut self, module: Module) -> Module {
```

Comment:
Can you merge `CommonJs` and `CommonJsWorker`?
Creating a new visitor is  really bad for binary size.